test/jdk/javax/net/ssl/TLS/TLSWontNegotiateDisabledCipherAlgos.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.net.ssl.*;
+import java.io.IOException;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.security.Security;
+import java.util.List;
+
+/*
+ * @test id=Server
+ * @bug 8301379
+ * @summary Verify that Java will not negotiate disabled cipher suites when the
```

Comment:
This test could be enhanced to test all of the other suites that are disabled. I think that might be a good idea, to centralize that. Although you could do it as a follow-on RFE.

Reply:
I added all of the disable ciphers that are listed in DisabledAlgorithms. Let me know if this isn't what you were thinking. Thanks!

Reply:
Yes, this is what I was thinking. There is another test `test/jdk/javax/net/ssl/ciphersuites/DisabledAlgorithms.java` that is very similar to this test. Can you compare these two tests and consider removing that test, or combining them if there are other things that test is testing that this one isn't?

Reply:
The tests look similar but are definitely testing different code paths. I didn't combine the tests because they're different enough in how they work that making them one test would be messier than having two tests. I did, however, relocate my new test to the same directory as `DisabledAlgorithms` and changed it to use the same list of disabled ciphers so when we need to add to the list, it's covered by both tests. Also, while there, I updated DisabledAlgorithms to use the SSL test template classes instead of the binary {key,trust}store files.

Reply:
Ok, I suggest adding a better comments to the `DisabledAlgorithms` test to note that they are different. The one difference I see is that `DisabledAlgorithms` is also testing that the disabled suites can still be used if the security properties are empty. Other than that, they look somewhat similar but using different templates for setting things up.